lib/idseq_utils/idseq_utils/batch_run_helpers.py

```diff
@@ -189,10 +189,23 @@ def _job_queue(provisioning_model: str):
         inputs["query_1"] = query_uris[1]
 
     wdl_input_uri = os.path.join(chunk_dir, f"{chunk_id}-input.json")
+    input_bucket, input_key = _bucket_and_key(wdl_input_uri)
+
     wdl_output_uri = os.path.join(chunk_dir, f"{chunk_id}-output.json")
+    output_bucket, output_key = _bucket_and_key(wdl_output_uri)
+
     wdl_workflow_uri = f"s3://idseq-workflows/{aligner}-{aligner_wdl_version}/{aligner}.wdl"
 
-    input_bucket, input_key = _bucket_and_key(wdl_input_uri)
+    # if this job fails we don't want to re-run chunks that have already been processed
+    #   the presence of the output file means the chunk has already been processed
+    try:
+        _s3_client.head_object(Bucket=output_bucket, Key=output_key)
+        log.info(f"skipping chunk, output already exists: {wdl_output_uri}")
+        return
+    except ClientError as e:
```

Comment:
I think we would want a `pass` here since we don't want to raise errors if the object doesn't exist, we just want to run the chunk. Is `ClientError` exhaustive as far as the errors that could come out of `.head_object`?

Reply:
This behavior passes if the error is a `404`. The current behavior raises all non-404 client errors. So if there were an unrelated error encountered while doing the check it would be raised. I can see maybe we would want to ignore those because we could recover by re-running the chunk but I think they will be notable enough that we would want to know and fix the issue.

Reply:
Sorry I saw `==` instead of `!=` there. Would errors from s3 throttling be raised here?

Reply:
Yeah it's confusing. Adding a comment. Throttle issues would be thrown here but we are already doing a put or two per chunk so the behavior is similar. We seem to mostly get those very rarely during the polling phase. I can add the adaptive strategy with lots of retries that we use for batch to the S3 client.

Reply:
Great thanks!